### PR TITLE
chore(ci): `check-all-features` on ready PRs

### LINKED ISSUE #1084: Investigate adding conditional CI
If it's possible to only run CI when a PR is reviewed, or a tag is added, or otherwise, we should try to do that for the more intensive CI checks like check features.

### PR DESCRIPTION
**Pull request by grtlr**
## Linked Issues

* Closes #1084.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,10 +39,12 @@ jobs:
     with: { os: ubuntu-latest, rust: nightly }
 
   clippy:
+    if: ${{ !github.event.pull_request.draft }}
```
Comment (Alex6323) [inline .github/workflows/ci.yml:42]: Did you read through the discussion here? There seemed to be some way of bypassing the CI if one merged too quickly. Did you check whether that can still happen? https://github.com/orgs/community/discussions/25722

Comment (grtlr) [reply .github/workflows/ci.yml:42]: From the discussion: https://github.com/orgs/community/discussions/25722#discussioncomment-3248917

This is exactly what we are doing, and it works nicely. I already tried it on this PR.

Comment (Alex6323) [reply .github/workflows/ci.yml:42]: I mean what the guy writes right below: https://github.com/orgs/community/discussions/25722#discussioncomment-3248918

**Review by Alex6323 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (grtlr) [reply .github/workflows/ci.yml:42]: I think this is an acceptable risk, as we also require at least one review.

**Pull request merged**